File: packages/sdk-python/steno_ai/webhooks.py
"""Webhook client — manage webhook subscriptions."""

from typing import List

from .client import HttpClient


class WebhookClient:
    def __init__(self, http: HttpClient):
        self._http = http

    def create(self, url: str, events: List[str], secret: str) -> dict:
        """Create a webhook subscription."""
        return self._http.request("POST", "/v1/webhooks", json={
            "url": url,
            "events": events,
            "secret": secret,
        })

    def list(self) -> list:
        """List all webhooks."""
        return self._http.request("GET", "/v1/webhooks")

    def delete(self, webhook_id: str) -> None:
        """Delete a webhook."""
        self._http.request("DELETE", f"/v1/webhooks/{webhook_id}")

File: packages/sdk-ts/src/clients/webhooks.ts
import type { HttpClient } from '../http.js';

export class WebhookClient {
  constructor(private readonly http: HttpClient) {}

  /** Create a webhook. */
  async create(options: { url: string; events: string[]; secret: string }): Promise<any> {
    return this.http.request('POST', '/v1/webhooks', options);
  }

  /** List all webhooks. */
  async list(): Promise<any> {
    return this.http.request('GET', '/v1/webhooks');
  }

  /** Delete a webhook by ID. */
  async delete(id: string): Promise<void> {
    return this.http.request('DELETE', `/v1/webhooks/${id}`);
  }
}
